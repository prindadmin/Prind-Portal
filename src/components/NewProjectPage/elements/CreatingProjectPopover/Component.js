import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
