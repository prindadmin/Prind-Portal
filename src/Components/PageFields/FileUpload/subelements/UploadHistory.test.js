import React from 'react';
import { shallow } from 'enzyme';

import Component from './UploadHistory'

var props;

beforeEach(() => {
  props = {
    projectId: "123",
    pageName: "design",
    fieldID: "1",
    details: [
      {
        uploadName: "Test File One.txt",
        uploadedBy: "Testy McTesterson",
        ver: "1",
        uploadedDateTime: "2021-04-05T08:38:32",
        proofLink: "https://prind.tech"
      }
    ]
  }
})


it('UploadHistory - Should render', () => {
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('UploadHistory - Should render without history', () => {
  props.details = []
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('UploadHistory - Should render without proof link', () => {
  props.details[0].proofLink = undefined
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('UploadHistory - Click download button', () => {
  const component = shallow(<Component {...props} />);
  const downloadButton = component.find('#download-box-container')
  downloadButton.simulate('click')
  expect(component).toMatchSnapshot();
});


it('UploadHistory - Click proof link', () => {
  const mockStopPropagation = jest.fn()
  const component = shallow(<Component {...props} />);
  const proofLink = component.find('#proof-link')
  proofLink.simulate('click', { stopPropagation: mockStopPropagation })
  expect(mockStopPropagation).toHaveBeenCalledTimes(1)
});


it('UploadHistory - Should render with no proof link', () => {
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});
