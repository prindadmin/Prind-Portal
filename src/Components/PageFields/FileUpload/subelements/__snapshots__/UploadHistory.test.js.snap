// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadHistory - Click download button 1`] = `
<div
  id="upload-history-container"
>
  <div
    className="element-title"
  >
    Upload History
  </div>
  <div
    className="details-table"
  >
    <h4>
      File Name: 
    </h4>
    <h4>
      Uploaded By: 
    </h4>
    <h4>
      Version: 
    </h4>
    <h4>
      Uploaded At: 
    </h4>
    <h4>
      Proof: 
    </h4>
    <h4>
      Download:
    </h4>
    <div
      key="uploadName-0"
    >
      Test File One.txt
    </div>
    <div
      key="uploadedBy-0"
    >
      Testy McTesterson
    </div>
    <div
      key="ver-0"
    >
      1
    </div>
    <div
      key="uploadDateTime-0"
    >
      Invalid Date
    </div>
    <div
      key="proof-0"
    >
      <a
        href="https://prind.tech"
        id="proof-link"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        Link to Proof
      </a>
    </div>
    <div
      key="downloadButton-0"
    >
      <Spinner
        size={16}
      />
    </div>
  </div>
</div>
`;

exports[`UploadHistory - Should render 1`] = `
<div
  id="upload-history-container"
>
  <div
    className="element-title"
  >
    Upload History
  </div>
  <div
    className="details-table"
  >
    <h4>
      File Name: 
    </h4>
    <h4>
      Uploaded By: 
    </h4>
    <h4>
      Version: 
    </h4>
    <h4>
      Uploaded At: 
    </h4>
    <h4>
      Proof: 
    </h4>
    <h4>
      Download:
    </h4>
    <div
      key="uploadName-0"
    >
      Test File One.txt
    </div>
    <div
      key="uploadedBy-0"
    >
      Testy McTesterson
    </div>
    <div
      key="ver-0"
    >
      1
    </div>
    <div
      key="uploadDateTime-0"
    >
      Invalid Date
    </div>
    <div
      key="proof-0"
    >
      <a
        href="https://prind.tech"
        id="proof-link"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        Link to Proof
      </a>
    </div>
    <div
      key="downloadButton-0"
    >
      <div
        id="download-box-container"
        onClick={[Function]}
      >
        <withRouter(Connect(DownloadBox))
          fieldID="1"
          fileVersionDetails={
            Object {
              "proofLink": "https://prind.tech",
              "uploadName": "Test File One.txt",
              "uploadedBy": "Testy McTesterson",
              "uploadedDateTime": "2021-04-05T08:38:32",
              "ver": "1",
            }
          }
          onDownloadFailure={[Function]}
          onDownloadSuccess={[Function]}
          pageName="design"
          projectId="123"
          size="small"
          style={
            Object {
              "alignItems": "flex-start",
              "border": "0px black solid",
              "height": "1.5em",
              "margin": "auto",
              "width": "1.5em",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`UploadHistory - Should render with no proof link 1`] = `
<div
  id="upload-history-container"
>
  <div
    className="element-title"
  >
    Upload History
  </div>
  <div
    className="details-table"
  >
    <h4>
      File Name: 
    </h4>
    <h4>
      Uploaded By: 
    </h4>
    <h4>
      Version: 
    </h4>
    <h4>
      Uploaded At: 
    </h4>
    <h4>
      Proof: 
    </h4>
    <h4>
      Download:
    </h4>
    <div
      key="uploadName-0"
    >
      Test File One.txt
    </div>
    <div
      key="uploadedBy-0"
    >
      Testy McTesterson
    </div>
    <div
      key="ver-0"
    >
      1
    </div>
    <div
      key="uploadDateTime-0"
    >
      Invalid Date
    </div>
    <div
      key="proof-0"
    >
      <a
        href="https://prind.tech"
        id="proof-link"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        Link to Proof
      </a>
    </div>
    <div
      key="downloadButton-0"
    >
      <div
        id="download-box-container"
        onClick={[Function]}
      >
        <withRouter(Connect(DownloadBox))
          fieldID="1"
          fileVersionDetails={
            Object {
              "proofLink": "https://prind.tech",
              "uploadName": "Test File One.txt",
              "uploadedBy": "Testy McTesterson",
              "uploadedDateTime": "2021-04-05T08:38:32",
              "ver": "1",
            }
          }
          onDownloadFailure={[Function]}
          onDownloadSuccess={[Function]}
          pageName="design"
          projectId="123"
          size="small"
          style={
            Object {
              "alignItems": "flex-start",
              "border": "0px black solid",
              "height": "1.5em",
              "margin": "auto",
              "width": "1.5em",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`UploadHistory - Should render without history 1`] = `
<div
  id="upload-history-container"
>
  <div
    className="element-title"
  >
    Upload History
  </div>
  <div>
    There are no previous version of this file
  </div>
</div>
`;

exports[`UploadHistory - Should render without proof link 1`] = `
<div
  id="upload-history-container"
>
  <div
    className="element-title"
  >
    Upload History
  </div>
  <div
    className="details-table"
  >
    <h4>
      File Name: 
    </h4>
    <h4>
      Uploaded By: 
    </h4>
    <h4>
      Version: 
    </h4>
    <h4>
      Uploaded At: 
    </h4>
    <h4>
      Proof: 
    </h4>
    <h4>
      Download:
    </h4>
    <div
      key="uploadName-0"
    >
      Test File One.txt
    </div>
    <div
      key="uploadedBy-0"
    >
      Testy McTesterson
    </div>
    <div
      key="ver-0"
    >
      1
    </div>
    <div
      key="uploadDateTime-0"
    >
      Invalid Date
    </div>
    <div
      key="proof-0"
    >
      Proof still processing
    </div>
    <div
      key="downloadButton-0"
    >
      <div
        id="download-box-container"
        onClick={[Function]}
      >
        <withRouter(Connect(DownloadBox))
          fieldID="1"
          fileVersionDetails={
            Object {
              "proofLink": undefined,
              "uploadName": "Test File One.txt",
              "uploadedBy": "Testy McTesterson",
              "uploadedDateTime": "2021-04-05T08:38:32",
              "ver": "1",
            }
          }
          onDownloadFailure={[Function]}
          onDownloadSuccess={[Function]}
          pageName="design"
          projectId="123"
          size="small"
          style={
            Object {
              "alignItems": "flex-start",
              "border": "0px black solid",
              "height": "1.5em",
              "margin": "auto",
              "width": "1.5em",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
